### ISSUE TITLE
[ExpandCollapse] Headings around collapsible sections

### ISSUE THREAD

**Issue by owestinTelus** [telus/tds-core#437]
## Description
- The advantage of collapsing content is to reduce scrolling. 
- Expanding the content is a choice to see it. 
- The button controlling the section offers a description of the hidden content. Essentially, the button is a heading providing an overview of the content.

## Consideration
- Should we wrap the collapse `<button>` in a heading? `<h#><button>Collapse Heading</button></h#>`
- The heading number would need to be decided depending on the content structure of the page
- Is there ever a need to use a collapse section without a heading? How are these being used on TELUS.com currently?

## Version Number of TDS
- [ ] v 0.30

**Comment by lucylist** [comment]
followed up with oskar on this issue. i have assigned it to him so that he can review it again and either make adjustments or close the issue if it's no longer a thing.

**Comment by owestinTelus** [comment]
### Update on the considerations in the original ticket

> We **should** wrap the collapse <button> in a heading. 
e.g.
`<h#><button>Collapse Heading</button></h#>`

The default for the component could be <h3> but the heading number would need to be modular in Site Builder so that it can be adjusted in context of the page.

**Comment by ryanoglesby08** [comment]
Hi @owestinTelus, are we able to say that ALL ExpandCollapses should ALWYAS use headings? Is seems that based on usage patterns I'm not sure it would ALWAYS be a heading element.

Maybe it should be configurable? With the option to not use a heading too?

**Comment by owestinTelus** [comment]
If we use ExpandCollapse for FAQs, we should create a definition list - that might be a separate component.

I can't think of a scenario where you wouldn't want a heading to describe the content in the ExpandCollapse. If it can be flexible enough to change it to a `<p>` or other tag that would be great. 

The default could still be `<h3>`

**Comment by ryanoglesby08** [comment]
I think if we were to implement something like this, we would make it more flexible, so the `header` could either be a simple string, or an arbitrary component/element. Then all the use cases you mentioned would be handled, you could use a paragraph, a heading, a definition list, or none of the above.

```jsx
<ExpandCollapse>
  <Panel header="Simple string">
    Some content...
  </Panel>
</ExpandCollapse>

<ExpandCollapse>
  <Panel header={<Heading level={3}>A heading component title</Heading>}>
    Some content...
  </Panel>
</ExpandCollapse>
```

**Comment by theetrain** [comment]
This is dependent on #605 

The main idea is to have heading aesthetics be consistent in all Panel headings, but with custom heading HTML elements.

**Comment by theetrain** [comment]
This was resolved in [@tds/core-expand-collapse@1.1.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-expand-collapse%401.1.0)
